Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -585,5 +642,32 @@ private string GetFullUrl(string url)
         {
             return url == null ? null : new Uri(new Uri(_baseUrl), url).ToString().Trim('/') + "/";
         }
+
+        /// <summary>
+        /// Create SourceRepository from given feed endpoint
+        /// </summary>
+        /// <param name="feedEndpoint">V2 or V3 feed endpoint</param>
+        /// <returns>SourceRepository object</returns>
+        private SourceRepository GetSourceRepository(string feedEndpoint)
+        {
+            IEnumerable<Lazy<INuGetResourceProvider, INuGetResourceProviderMetadata>> providers = _container.GetExports<INuGetResourceProvider, INuGetResourceProviderMetadata>();
```

Comment:
i am trying to move "providers" property out, and use MEF to import it. so that MEF will take care of the liftcycle